pandas/core/series.py

```diff
@@ -4475,8 +4475,17 @@ def set_axis(self, labels, axis: Axis = 0, inplace: bool = False):
         optional_labels=_shared_doc_kwargs["optional_labels"],
         optional_axis=_shared_doc_kwargs["optional_axis"],
     )
-    def reindex(self, index=None, **kwargs):
-        return super().reindex(index=index, **kwargs)
+    def reindex(self, *args, **kwargs) -> Series:
+        if len(args) > 1:
+            raise TypeError("Only one positional argument ('index') is allowed")
+        if args:
```

Comment:
Is there a test when ``args`` is not truthy?
It seems that this check ``if args`` have to be carried out before ``if len(args) > 1`` otherwise the latter one will fail.
I suggest that you add a test when ``args`` are not supplied (unless it does already exist in the code base).

Reply:
Hey @ivanovmg - `args` will be a tuple, so if the user doesn't pass `args`, then it should just be an empty tuple and `len(args)` will be 0. I've added a test in which reindex is used without positional arguments anyway, thanks :+1: